base_layer/wallet/src/transaction_service/protocols/transaction_coinbase_monitoring_protocol.rs

```diff
@@ -154,275 +154,427 @@ where TBackend: TransactionBackend + 'static
-                    completed_tx.tx_id,
-                )))
-                .map_err(|e| {
-                    trace!(
-                        target: LOG_TARGET,
-                        "Error sending event, usually because there are no subscribers: {:?}",
+                let _ = self
+                    .resources
+                    .event_publisher
+                    .send(Arc::new(TransactionEvent::TransactionMinedRequestTimedOut(self.tx_id)))
+                    .map_err(|e| {
+                        trace!(
+                            target: LOG_TARGET,
+                            "Error sending event, usually because there are no subscribers: {:?}",
+                            e
+                        );
                         e
-                    );
-                    e
-                });
+                    });
+            }
         }
     }
 
-    async fn handle_base_node_response(
+    async fn query_coinbase_transaction(
         &mut self,
-        tx_id: TxId,
-        response: proto::BaseNodeServiceResponse,
-    ) -> Result<BaseNodeResponseType, TransactionServiceProtocolError>
+        signature: Signature,
+        client: &mut BaseNodeWalletRpcClient,
+    ) -> Result<(bool, Option<u64>), TransactionServiceProtocolError>
     {
-        let mut returned_ouputs: Vec<tari_core::proto::types::TransactionOutput> = Vec::new();
-        match response.response {
-            Some(BaseNodeResponseProto::TransactionOutputs(outputs)) => returned_ouputs = outputs.outputs,
-            Some(BaseNodeResponseProto::ChainMetadata(metadata)) => {
-                if let Some(tip) = metadata.height_of_longest_chain {
-                    return if tip > self.block_height {
-                        Ok(BaseNodeResponseType::ChainMetadata(false))
-                    } else {
-                        Ok(BaseNodeResponseType::ChainMetadata(true))
-                    };
-                }
-            },
-            _ => {
-                return Ok(BaseNodeResponseType::Other);
+        trace!(
+            target: LOG_TARGET,
+            "Querying status for coinbase transaction (TxId: {})",
+            self.tx_id,
+        );
+        let response = match client.transaction_query(signature.into()).await {
+            Ok(r) => match TxQueryResponse::try_from(r) {
+                Ok(r) => r,
+                Err(_) => {
+                    trace!(
+                        target: LOG_TARGET,
+                        "Could not convert proto TxQueryResponse for coinbase transaction (TxId: {})",
+                        self.tx_id,
+                    );
+                    return Ok((false, None));
+                },
             },
-        }
-
-        let completed_tx = match self.resources.db.get_completed_transaction(tx_id).await {
-            Ok(tx) => tx,
             Err(e) => {
-                error!(
+                info!(
```

Comment:
If this is important enough to not be `debug` then shouldn't it be `warn` instead of `info` ?